src/Pages/AccountPage/SignUp.js

```diff
@@ -0,0 +1,224 @@
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+
+export default function SignUp({ changeType, changeModalOpen }) {
+  const [dlList, setDlList] = useState([]);
+  const [circleClickArr, setcircleClickArr] = useState([]);
+  const [isCircleAll, setIsCircleAll] = useState(false);
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  useEffect(() => {
+    setIsCircleAll(
+      circleClickArr.length > 0 && circleClickArr.every(isClicked => isClicked)
+    );
+  }, [circleClickArr]);
+
+  const fetchData = () => {
+    fetch('/data/SignUp.json')
+      .then(res => res.json())
+      .then(data => {
+        setDlList(data);
+        setcircleClickArr(Array.from({ length: data.length }, () => false));
+      });
+  };
+
+  const selectCircleClick = index => {
+    setcircleClickArr(
+      circleClickArr.map((circle, i) => (i === index ? !circle : circle))
+    );
+  };
+
+  const isAllCircleClick = () => {
+    setIsCircleAll(!isCircleAll);
+    setcircleClickArr(
+      circleClickArr.map(circle => (isCircleAll ? !circle : true))
+    );
+  };
+
+  return (
+    <Signup>
+      <Header>
+        <span>약관동의</span>
+        <XIcon
+          onClick={() => changeModalOpen(false)}
+          className="fas fa-times"
+        />
+      </Header>
+      <Main>
+        {dlList.length > 0 &&
+          dlList.map((dl, index) => {
+            return (
+              <Dl key={dl.id}>
+                <DT onClick={() => selectCircleClick(index)}>
+                  <span>
+                    {dl.title}
+                    <strong>
+                      {index === dlList.length - 1 ? '(선택)' : '(필수)'}
+                    </strong>
+                  </span>
+                  <CircleIcon
+                    blue={circleClickArr[index]}
+                    className="far fa-check-circle"
+                  />
+                </DT>
+                <DD isExist={!!dl.descContent}>
+                  <strong className="title">{dl.descTitle}</strong>
+                  <p>{dl.descContent}</p>
+                </DD>
+              </Dl>
+            );
+          })}
+
+        <AgreeAll>
+          <span>
+            모두 동의합니다.<strong>(필수)</strong>
+          </span>
+          <CircleIcon
+            blue={isCircleAll}
```

Comment:
```suggestion
            isChecked={isCircleAll}
```
생각해보니 blue보다는 isChecked가 보다 좋아보이네요

Reply:
넵!@